src/Conversion/ONNXToTorch/NN/FlattenOp.cpp

```diff
@@ -0,0 +1,161 @@
+}
+
+class ONNXFlattenOpToTorchLowering : public ConversionPattern {
+public:
+  ONNXFlattenOpToTorchLowering(TypeConverter &typeConverter,
+		  MLIRContext *ctx)
+      : ConversionPattern( typeConverter,
+		      ::mlir::ONNXFlattenOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+
+    Location loc = op->getLoc();
+    mlir::MLIRContext *context = op->getContext();
+    ONNXFlattenOp op1 = llvm::dyn_cast<ONNXFlattenOp>(op);
+
+    Value input = op1.input();
+    auto axisValue = op1.axis();       // ::mlir::IntegerAttr
+
+    auto inputShape = input.getType().cast<ShapedType>().getShape();
+    int64_t inputRank = inputShape.size();
+
+    TensorType resultTensorType =
+	    op->getResult(0).getType().cast<TensorType>();
+    auto resultType = Torch::ValueTensorType::get(op1.getContext(),
+          resultTensorType.getShape(), resultTensorType.getElementType());
+
+    TensorType inputTensorType  = input.getType().cast<TensorType>();
+    auto inputType =
+	 Torch::ValueTensorType::get(context, inputTensorType.getShape(),
+                    inputTensorType.getElementType());
+    auto inputTensor =
+	  rewriter.create<torch::TorchConversion::FromBuiltinTensorOp>(loc,
+                    inputType, input);
+    Value result = inputTensor;
+
+    // if axisValue is negative
+    if (axisValue < 0)
+      return op->emitError("negative axis not supported");
+    
+    /***************************************************************/
+    // if axisValue is 0 or 1, we need to flatten once.
+    // i) if axisValue is 0, start_dim will be -1 (axisValue -1 ) and
+    //    end_dim will be 0.
+    // ii) if axisValue is 1, start_dim will be 0 (axisValue - 1) and
+    //    end_dim will 0. Flatten from 0 to 0  flatten was not emitted.
+    //
+    // Because of these reasons, need to emit flatten once with
+    //   start_dim = axisValue and end_dim = inputRank.
+    //
+    // If axisValue is more than 1, emit the flatten two times like below.
+    //    a) Flattening is about bringing the flattened zone into a
+    //       single dimensional.
+    //    b) torch flatten flattens the region between (start-dim
+    //       and end-dim).
+    //    c) onnx flattening is about creating a 2-D vector,
+    //       the first dim consisting of values till axis-1, the second
+    //       dim consisting of values from axis to end.
+    //
+    // What we feel is that there will be two steps required -
+    //
+    //  1) flatten the region from 0 position to axis - 1.
+    //  2) flatten the region from axisValue to inputRank( last dimension).
+    /********************************************************************/
+
+    if (axisValue > 1) {
+      // flatten the region upto axis-1.
+      result = createAtenFlattenOp (rewriter, loc, result, resultType, 0, 
```

Comment:
I think the type of this first flatten should not be the final `resultType`. Given an input shape of `2x3x5x7` and an axis of 2, this should generate 2 flatten ops with the following types (pseudocode):
```mlir
%x = "flatten" %inpt, 0, 1 : !torch.vtensor<[2,3,5,7],f32>, !torch.int, !torch.int -> !torch.vtensor<[6,5,7],f32>
%y = "flatten" %x, 1, -1 : !torch.vtensor<[6,5,7],f32>, !torch.int, !torch.int -> !torch.vtensor<[6,35],f32>
```
Currently what's generated uses the final type `[6,35]` as the intermediate type `[6,5,7]`, which makes the aten ill-typed (at least I think, but no tool I used produced a type error...)